pkg/apis/core/validation/shoot.go

```diff
@@ -865,6 +865,19 @@ func validateKubernetes(kubernetes core.Kubernetes, dockerConfigured, shootHasDe
 			}
 		}
 
+		if kubeAPIServer.AccessControl != nil {
+			// Action available, but no Source
+			if kubeAPIServer.AccessControl.Action != nil && kubeAPIServer.AccessControl.Source == nil {
+				allErrs = append(allErrs, field.Required(fldPath.Child("kubeAPIServer", "accessControl", "action"), "is required when access control action enabled"))
```

Comment:
Maybe I got it wrong, but it seems this should rather complain about the source and not the action.
```suggestion
				allErrs = append(allErrs, field.Required(fldPath.Child("kubeAPIServer", "accessControl", "source"), "is required when access control action enabled"))
```